scribble.go

```diff
@@ -199,6 +199,39 @@ func (d *Driver) ReadAll(collection string) ([]string, error) {
 	return records, nil
 }
 
+// List all records from a collection; this is returned as a slice of strings because
+// there is no way of knowing what type the record is.
+func (d *Driver) List(collection string) ([]string, error) {
+
+	// ensure there is a collection to read
+	if collection == "" {
+		return nil, fmt.Errorf("missing collection - unable to record location")
+	}
+
+	//
+	dir := filepath.Join(d.dir, collection)
+
+	// check to see if collection (directory) exists
+	if _, err := stat(dir); err != nil {
+		return nil, nil
```

Comment:
If we're doing a return here at all, best to return some sort of info (nil, err).

Reply:
Originally I gave here (nil, err), but in practice it turned out to be a issue. So I decided to do it this way.

For a momentary reflection, I come to the conclusion that at the application level I will always ignore err from List(), so we can actually return this error there and if someone is interested in it, will be able to read it.